src/main/java/org/killbill/billing/codegen/languages/KillbillJavaGenerator.java

```diff
@@ -196,46 +207,74 @@ public Map<String, Object> postProcessModels(Map<String, Object> objs) {
-
         final List<Map<String, String>> imports = (List<Map<String, String>>) objs.get("imports");
-
-
         for (CodegenOperation op : operations) {
             final ExtendedCodegenOperation ext = new ExtendedCodegenOperation(op);
             extOperations.add(ext);
-            if (ext.isReturnModelRefContainer) {
-                addImportIfRequired(imports, String.format("org.killbill.billing.client.model.%s", ext.returnType));
-            }
-            if (ext.isBodyModelRefContainer) {
-                addImportIfRequired(imports, String.format("org.killbill.billing.client.model.%s", ext.bodyParam.dataType));
-            }
-            if (ext.isStream) {
-                addImportIfRequired(imports, "java.io.InputStream");
-                addImportIfRequired(imports, "java.io.OutputStream");
-                addImportIfRequired(imports, "java.net.http.HttpResponse");
-            }
-            if (ext.isListContainer) {
-                addImportIfRequired(imports, "java.util.List");
-            } else if (ext.isMapContainer) {
-                addImportIfRequired(imports, "java.util.Map");
+            addRequiredImports(ext, imports);
+            convertToExtendedCodegenParam(ext, imports);
+            if (shouldAddDateTimeMethod(op)) {
+                addImportIfRequired(imports, "org.joda.time.DateTime");
+                final ExtendedCodegenOperation ext2 = new ExtendedCodegenOperation(op);
+                addRequiredImports(ext2, imports);
+                List<CodegenParameter> allParams = ext2.allParams;
+                for (CodegenParameter parameter : allParams) {
+                    if (isDateParameter(parameter)) {
+                        parameter.dataFormat = "date-time";
+                        parameter.isDate = false;
+                        parameter.isDateTime = true;
+                        parameter.dataType = "DateTime";
+                    }
+                }
+                convertToExtendedCodegenParam(ext2, imports);
+                extOperations.add(ext2);
             }
-
-            // Unfortunately those lists contain different objects -- so a query param from all param is really a different
-            // java object and if needed needs to be converted for each list
-            ext.allParams = convertToExtendedCodegenParam(ext.allParams, imports);
-            ext.bodyParams = convertToExtendedCodegenParam(ext.bodyParams, imports);
-            ext.pathParams = convertToExtendedCodegenParam(ext.pathParams, imports);
-            ext.queryParams = convertToExtendedCodegenParam(ext.queryParams, imports);
-            ext.formParams = convertToExtendedCodegenParam(ext.formParams, imports);
         }
         operationsMap.put("operation", extOperations);
-
         return objs;
     }
+     
+    private void addRequiredImports(ExtendedCodegenOperation ext, List<Map<String, String>> imports) {
```

Comment:
Perhaps rename as `addAllImportsIfRequired` since this call a bunch of times `addImportIfRequired` ?

Reply:
Fixed